### PR TITLE
vapoursynth: update to 54

### PR DESCRIPTION
**Pull request by WhiredPlanck**
Topic Description
-----------------

vapoursynth: update to 54

Package(s) Affected
-------------------

- `vapoursynth`
- `zimg`

Security Update?
----------------

No

Build Order
-----------

`tesseract` ( #3350  ), `zimg` -> `vapoursynth`

Architectural Progress
----------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

**Comment by WhiredPlanck** [comment]
~~Applying patch `autobuild/patches/0001-Fix-compilation-for-non-x86-systems.patch` failed, and help wanted.~~

**UPDATE**: patches are unnecessary any more because they are merged into the source by upstream.

**Comment by WhiredPlanck** [comment]
LGTM (vapoursynth, amd64)
图片

File: extra-multimedia/vapoursynth/autobuild/beyond [hunk 1]
```diff
@@ -0,0 +1 @@
+sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
```
Comment (MingcongBai) [inline extra-multimedia/vapoursynth/autobuild/beyond]: This is unnecessary for AOSC OS. Before you copy from Arch Linux, think carefully about what each line does.

File: extra-multimedia/vapoursynth/autobuild/prepare [hunk 2]
```diff
@@ -1,4 +1,5 @@
-export CXXFLAGS="${CXXFLAGS/-fdeclone-ctor-dtor/}"
+#export CXXFLAGS="${CXXFLAGS/-fdeclone-ctor-dtor/}"
```
Comment (MingcongBai) [inline extra-multimedia/vapoursynth/autobuild/prepare]: Remove all lines if you didn't need it, this is Git, don't comment out code.

File: extra-multimedia/vapoursynth/autobuild/prepare [hunk 3]
```diff
@@ -1,4 +1,5 @@
...
+export LDFLAGS="$LDFLAGS -lpthread"
```
Comment (MingcongBai) [inline extra-multimedia/vapoursynth/autobuild/prepare:2]: Missing `abinfo` here.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-multimedia/vapoursynth/autobuild/beyond (hunk 1 above)
Comment (WhiredPlanck) [reply extra-multimedia/vapoursynth/autobuild/beyond]: This is indeed something that I didn't think about carefully, so I won't take it as an example.

Comment (WhiredPlanck) [reply extra-multimedia/vapoursynth/autobuild/beyond]: Dropped this in 145529b.

File: extra-multimedia/vapoursynth/autobuild/prepare (hunk 2 above)
Comment (WhiredPlanck) [reply extra-multimedia/vapoursynth/autobuild/prepare]: Resolved in 14843fe.

File: extra-multimedia/vapoursynth/autobuild/prepare (hunk 3 above)
Comment (WhiredPlanck) [reply extra-multimedia/vapoursynth/autobuild/prepare:2]: Resolved in 14843fe.

File: extra-libs/zimg/autobuild/defines [hunk 4]
```diff
@@ -2,10 +2,7 @@ PKGNAME=zimg
...
+PKGBREAK="vapoursynth<=48"
 
-AUTOTOOLS_AFTER__AMD64=" \
-                 ${AUTOTOOLS_AFTER} \
-                 --enable-x86simd"
-AUTOTOOLS_AFTER__I486=" \
-                 ${AUTOTOOLS_AFTER} \
-                 --enable-x86simd"
+AUTOTOOLS_AFTER="--prefix=/usr \
+            --enable-simd"
```
Comment (MingcongBai) [inline extra-libs/zimg/autobuild/defines]: Align your lines.

File: extra-libs/zimg/autobuild/defines [hunk 5]
```diff
@@ -2,10 +2,7 @@ PKGNAME=zimg
 PKGSEC=libs
 PKGDEP="gcc-runtime"
 PKGDES="Scaling, colorspace conversion, and dithering library"
+PKGBREAK="vapoursynth<=48"
```
Comment (MingcongBai) [inline extra-libs/zimg/autobuild/defines]: The newest version is obviously not 48. Check with `apt list vapoursynth`.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-libs/zimg/autobuild/defines (hunk 5 above)
Comment (WhiredPlanck) [reply extra-libs/zimg/autobuild/defines]: Sorry for that, I took it for granted that put the VER here but not append the REL, I'll fix it soon. Thanks~

Comment (WhiredPlanck) [reply extra-libs/zimg/autobuild/defines]: Resolved in 4d084c8.

File: extra-libs/zimg/autobuild/defines (hunk 4 above)
Comment (WhiredPlanck) [reply extra-libs/zimg/autobuild/defines]: Resolved in 4d084c8.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**